src/components/pages/ResultQuizPage/ResultQuizPage.tsx

```diff
@@ -31,20 +31,19 @@ const ResultQuizPage = () => {
         navigate(START);
     };
 
-    const saveStatistics = () => {
+    const saveStatistics = useCallback(() => {
         dispatch(addCurrentQuizResultToStatistics({
             questions: questions.length,
             correctAnswers,
             category,
             difficulty, 
             type
         }));
-    };
+    }, [correctAnswers, questions, category, difficulty, type, dispatch]);
 
     useEffect(() => {
-        console.log('Mounting result pade ande geting quiz results');
-        saveStatistics()
-    });
+        saveStatistics();
+    }, [saveStatistics]);
```

Comment:
I see what you want to reach
```tsx
    useEffect(() => {
dispatch(addCurrentQuizResultToStatistics({
            questions: questions.length,
            correctAnswers,
            category,
            difficulty, 
            type
        }));
    }, [correctAnswers, questions, category, difficulty, type, dispatch]));
```
this is more clear
in your case no need to create a callback :)